tests/Integration/ComposerEventHandlerTest.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Config\Tests\Integration;
+
+
+use PHPUnit\Framework\TestCase;
+use function dirname;
+use function in_array;
+
+//use function PHPUnit\Framework\directoryExists;
+
+final class ComposerEventHandlerTest extends TestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+        $workingDirectory = $this->getWorkingDirectory();
+        $this->exec("rm -rf $workingDirectory/*");
+        file_put_contents($workingDirectory . '/composer.json',
+            <<<TXT
+{
+    "name": "yiisoft/testpackage",
+    "type": "library",
+    "minimum-stability": "dev",
+    "require": {
+        "yiisoft/config": "*"
+    },
+    "repositories": [
+        {
+            "type": "path",
+            "url": "../../"
+        },
+        {
+            "type": "path",
+            "url": "../Packages/first-vendor/first-package",
+            "options": {
+              "symlink": false
+            }
+        }
+    ]
+}
+TXT
+        );
+        $this->execComposer('install');
+    }
+
+    protected function tearDown(): void
+    {
+        parent::tearDown();
+        $workingDirectory = $this->getWorkingDirectory();
+        $this->exec("rm -rf $workingDirectory/*");
```

Comment:
Won't work with Windows. Let's require-dev yiisoft/files and use `FileHelper::removeDirectory()`.